File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/facade/user/UserHeartFacade.java
package com.cakk.domain.mysql.facade.user;

import lombok.RequiredArgsConstructor;

import com.cakk.domain.mysql.annotation.DomainFacade;
import com.cakk.domain.mysql.entity.cake.Cake;
import com.cakk.domain.mysql.entity.shop.CakeShop;
import com.cakk.domain.mysql.entity.user.User;

@RequiredArgsConstructor
@DomainFacade
public class UserHeartFacade {

	public void heartCake(final User user, final Cake cake) {
		if (!cake.isHeartedBy(user)) {
			user.heartCake(cake);
		} else {
			user.unHeartCake(cake);
		}
	}

	public void heartCakeShop(final User user, final CakeShop cakeShop) {
		if (!cakeShop.isHeartedBy(user)) {
			user.heartCakeShop(cakeShop);
		} else {
			user.unHeartCakeShop(cakeShop);
		}
	}
}

File: cakk-core/src/main/kotlin/com/cakk/core/facade/user/UserHeartFacade.kt
package com.cakk.core.facade.user

import com.cakk.core.annotation.DomainFacade
import com.cakk.infrastructure.persistence.entity.cake.CakeEntity
import com.cakk.infrastructure.persistence.entity.shop.CakeShopEntity
import com.cakk.infrastructure.persistence.entity.user.UserEntity

@DomainFacade
class UserHeartFacade {

    fun heartCake(userEntity: UserEntity, cake: CakeEntity) {
        if (!cake.isHeartedBy(userEntity)) {
            userEntity.heartCake(cake)
        } else {
            userEntity.unHeartCake(cake)
        }
    }

    fun heartCakeShop(userEntity: UserEntity, cakeShop: CakeShopEntity) {
        if (!cakeShop.isHeartedBy(userEntity)) {
            userEntity.heartCakeShop(cakeShop)
        } else {
            userEntity.unHeartCakeShop(cakeShop)
        }
    }
}
